scripts/process/process_map.R

```diff
@@ -0,0 +1,83 @@
+
+#' take map arguments and return a projected sp object
+#' 
+#' @param \dots arguments passed to \code{\link[maps]{map}} excluding \code{fill} and \code{plot}
+#' 
+library(maps)
+library(sp)
+proj.string <- "+proj=laea +lat_0=45 +lon_0=-100 +x_0=0 +y_0=0 +a=6370997 +b=6370997 +units=m +no_defs"
+to_sp <- function(...){
+  library(maptools)
+  library(maps)
+  map <- maps::map(..., fill=TRUE, plot = FALSE)
+  IDs <- sapply(strsplit(map$names, ":"), function(x) x[1])
+  map.sp <- map2SpatialPolygons(map, IDs=IDs, proj4string=CRS("+proj=longlat +datum=WGS84"))
+  map.sp.t <- spTransform(map.sp, CRS(proj.string))
+  return(map.sp.t)
+}
+
+#' create the sp object 
+#'
+#'@param viz the vizlab object 
+process.state_map <- function(viz){
+  library(sp)
+  conus <- to_sp('state')
+  HI <- to_sp("world", "USA:hawaii")
+  AK <- to_sp("world", "USA:alaska")
+  PR <- to_sp("world", "Puerto Rico")
+  
+  # thanks to Bob Rudis (hrbrmstr):
+  # https://github.com/hrbrmstr/rd3albers
+  
+  # -- if moving any more states, do it here: --
+  alaska <- shift_sp(AK, 0.33, shift = c(80,-450), rotate=-50, proj.string = proj4string(conus), row.names = 'alaska') 
```

Comment:
Any desire to try to follow the list workflow I tried here:\

http://dev-owi.usgs.gov-website.s3-website-us-west-2.amazonaws.com/blog/usMap/